logger.go

```diff
@@ -45,3 +46,28 @@ type ZapCompatibleLogger interface {
 	Fatalf(template string, args ...interface{})
 	Fatalw(msg string, keysAndValues ...interface{})
 }
+
+// Sublogger creates a sublogger from the given ZapCompatibleLogger instance.
+// This function uses reflection to dynamically create a sublogger from the provided logger by
+// calling its `Sublogger` method if it is an RDK logger, or its `Named` method if it is a Zap logger.
+// If neither method is available, it logs a debug message and returns the original logger.
+func Sublogger(inp ZapCompatibleLogger, subname string) ZapCompatibleLogger {
+	typ := reflect.TypeOf(inp)
+	sublogger, ok := typ.MethodByName("Sublogger")
+	if !ok {
+		sublogger, ok = typ.MethodByName("Named")
+		if !ok {
+			inp.Debugf("could not create sublogger from logger of type %s, returning self", typ.String())
+			return inp
+		}
+	}
+
+	ret := sublogger.Func.Call([]reflect.Value{reflect.ValueOf(inp), reflect.ValueOf(subname)})
```

Comment:
This panics if the function inputs aren't what we expect. I'm not sure how defensive we want to be here.

If we want to just `defer recover` and `log + return inp`, that's good enough. I don't realistically expect this error case to happen.

If we want to try to catch this before* calling the `sublogger.Func`, I think this starts with the `subloggerMethod.Type.NumIn()` call. Presumably we expect that to be `2` (the receiver argument + the sublogger name). And `subloggerMethod.Type.In(idx)` to check the types of the inputs.

Reply:
Good catch; I'm also fine with `defer` + `recover`.

Reply:
`defer` + `recover` also good with me and safer